packages/overlay/src/integrations/sentry/components/traces/spans/SpanTree.tsx

```diff
@@ -21,10 +23,29 @@ export default function SpanTree({
   spanNodeWidth: number;
   setSpanNodeWidth?: (val: number) => void;
 }) {
+  const { query, matchesQuery, showOnlyMatched } = useSearch();
+
+  const filteredTree = useMemo(() => {
+    if (!query) return tree;
```

Comment:
Early return here with `showOnlyMatched` would make it easier to follow.
```suggestion
    if (!query || !showOnlyMatched) return tree;
```